JSDemos/Demos/DataGrid/StateManagement/jQuery/index.js

```diff
@@ -0,0 +1,122 @@
+$(function () {
+    var URL = "https://js.devexpress.com/Demos/Mvc/api/DataGridWebApi";
+
+    var loadPanel = $("#loadPanel").dxLoadPanel({
+        position: {
+            of: "#gridContainer"
+        },
+        visible: false
+    }).dxLoadPanel("instance");
+
+    loadPanel.show();
+    sendRequest(URL + "/Orders?skip=700")
+        .always(() => { loadPanel.hide(); })
+        .then((data) => {
+            dataGrid.option("dataSource", data);
+        });
+
+    var dataGrid = $("#gridContainer").dxDataGrid({
+        keyExpr: "OrderID",
+        showBorders: true,
+        dataSource: [],
+        editing: {
+            mode: "row",
+            allowAdding: true,
+            allowUpdating: true,
+            allowDeleting: true
+        },
+        repaintChangesOnly: true,
+        onOptionChanged: function(e) {
+            if(e.name === "editing") {
+                var editRowKey = e.component.option("editing.editRowKey"),
+                    changes = e.component.option("editing.changes");
+
+                $("#editRowKey").text(editRowKey === null ? "null" : editRowKey);
+
+                changes = changes.map((change) => { 
+                    return {
+                        type: change.type,
+                        key: change.type !== "insert" ? change.key : undefined,
+                        data: change.data
+                    };
+                 });
+
+                $("#changes").text(JSON.stringify(changes, null, " "));      
+            }
+        },
+        onSaving: function(e) {
+            var change = e.changes[0];
+
+            e.cancel = true;
+
+            if (change) {
```

Comment:
Move `e.cancel = true` to `if(change) {` block because save button should work if change does not exist.